dockerfiles/jaeger/all-in-one/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# This Dockerfile tracks jaegertracing/all-in-one, but includes more debugging tools and runs as a
+# non-root user. It requires JAEGER_VERSION to be set as an argument to build.
+ARG JAEGER_VERSION
+FROM jaegertracing/all-in-one:$JAEGER_VERSION as base
+
+FROM alpine:3.11.5
+RUN apk update
+RUN apk add bash curl
+
+COPY --from=base /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+COPY --from=base /go/bin/all-in-one-linux  /go/bin/all-in-one-linux
+COPY --from=base /etc/jaeger/sampling_strategies.json /etc/jaeger/sampling_strategies.json
+
+RUN adduser -S -u 10001 jaeger
+RUN chown jaeger /go/bin/all-in-one-linux
+RUN chown jaeger /etc/jaeger/sampling_strategies.json
+USER jaeger
+
+# Agent zipkin.thrift compact
+EXPOSE 5775/udp
+
+# Agent jaeger.thrift compact
+EXPOSE 6831/udp
+
+# Agent jaeger.thrift binary
+EXPOSE 6832/udp
+
+# Agent config HTTP
+EXPOSE 5778
+
+# Collector HTTP
+EXPOSE 14268
+
+# Collector gRPC
+EXPOSE 14250
+
+# Web HTTP
+EXPOSE 16686
+
+VOLUME ["/tmp"]
```

Comment:
Will this Docker image ever be used in non-Kubernetes environments? If yes, you MUST create `/tmp` within the image with the desired permissions or else the container will not be able to write to `/tmp` in certain environments. See https://sourcegraph.com/github.com/sourcegraph/sourcegraph@730f31fb13b8380ed02fd01fe4a8583e43b029d4/-/blob/cmd/precise-code-intel/api-server/Dockerfile#L30-37